script/github-actions/select-cypress-tests.js

```diff
@@ -148,15 +151,29 @@ function selectedTests(graph, pathsOfChangedFiles) {
     tests.push(...glob.sync(selectedTestsPattern));
   });
 
-  // Always run the tests in src/platform
-  const defaultTestsPattern = path.join(
-    __dirname,
-    '../..',
-    'src/platform',
-    '**/tests/**/*.cypress.spec.js?(x)',
-  );
+  // Only run the mega menu test for changed apps builds,
+  // otherwise run all tests in src/platform for full builds
+  if (IS_CHANGED_APPS_BUILD) {
+    const megaMenuTestPath = path.join(
+      __dirname,
+      '../..',
+      'src/platform/site-wide/mega-menu/tests/megaMenu.cypress.spec.js',
+    );
+
+    // Ensure changed apps have URLs to run header test on
+    if (APPS_HAVE_URLS && fs.existsSync(megaMenuTestPath))
```

Comment:
This will need to be updated to account for the static-pages app once it's ready to be put on the allow-list, since it has no `rootUrl` but should run the header test. No need to add that logic now since it's not needed yet.

Reply:
Good question. Since changes to an app would mostly affect the header on just that app's page, we would need to use a URL that uses that app's asset. Because an app without a URL would be the only app asset that gets built, no other app URL would probably work for the header test.

One idea could be to add an option to the manifest file of apps without URLs that would let the webpack build know that it should make a test page for that app to test the header on. Using the [proxy-rewrite](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/proxy-rewrite/manifest.json) app as an example, adding something like 

``` json
"e2eUrl": "/proxy-rewrite"
```

to it's manifest file could be used to let the webpack build know to make a scaffold page at `/proxy-rewrite` that uses the `proxy-rewrite` entry. There would probably be no app components that get loaded on that page, but the header would still work since we're referencing the app's assets. We would just need to add that functionality to the [generateHtmlFiles](https://github.com/department-of-veterans-affairs/vets-website/blob/master/config/webpack.config.js#L156) function.

Reply:
That all makes sense, can't run the header test for a different app if that app isn't built, haha. It could be fine to just skip the header test in cases where there's no page, if that adds too much complexity. In that case we could wait and see if a page-less app ever breaks the header and add the test in that case. Maybe it will never happen!

Reply:
I agree, if we ever come across a case where that happens, we can always update the process to handle that. The only app without a URL I could see that happening with is static-pages, but we already know we'll need to run the test on the homepage when it's ready to be added to the allow-list. That'd be ironic if it ends up being a page-less app that breaks the header haha